docs/src/components/templates.tsx

```diff
@@ -117,6 +117,18 @@ function useLastTruthyValue<T>(val: T): T {
 }
 
 export default function Templates() {
+  /* As the <TemplateList /> component uses useSearchParams it must be wrapped in
+   * a Suspense boundary to prevent the whole tree being rendered client-side only.
+   * See: https://nextjs.org/docs/app/api-reference/functions/use-search-params#static-rendering
```

Comment:
Nice find! TIL 🙌 I really need to up my app dir game.

Reply:
This is such a foot-gun that hopefully Next throws a warning or something in future to let you know that using this hook will break ssr on your entire tree. 

I only found it by brute forcing a solution